sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -649,6 +649,24 @@ def test_radius_neighbors_boundary_handling():
         assert_array_equal(results[0], [0, 1])
 
 
+def test_radius_neighbors_returns_array_of_objects():
+    """regression for issue #16036
+    """
+    X = csr_matrix(np.ones((4, 4)))
+    X.setdiag([0, 0, 0, 0])
+
+    nbrs = neighbors.NearestNeighbors(radius=0.5, algorithm='auto',
+                                      leaf_size=30, metric='precomputed',
+                                      metric_params=None, p=2,
```

Comment:
We can remove `metric_params`, `p`, and `n_jobs` because these are the default values.

Reply:
Done